File: PYTHON/menstrual_app.py
import function_menstrual_app
						
print(function_menstrual_app.display_welcome_page())
user_input = int(input('Enter 1: '))
if user_input == 1:
	print(function_menstrual_app.menuPage())
	menu_input = int(input('Enter an option: '))
	match menu_input:
		case 1: 
			print(function_menstrual_app.in_app_display())
			flowDate_input = int(input('Enter the last day you saw your period: '))
			flow_check = function_menstrual_app.calculate_flow_date(flowDate_input)
			print("Your next flow date is ",flow_check, " days time")
		case 2:
			print(function_menstrual_app.in_app_display())
			safePeriod_input = int(input('Enter the last day you saw your period: '))
			pre_ovulation_constant = 7
			post_ovulation_constant = 15
			post_ovulation_constant2 = 28
			if safePeriod_input > 0 and safePeriod_input <= pre_ovulation_constant:
				calculate_pre_ovulation = pre_ovulation_constant -  safePeriod_input
				print("You are on your safe period")
				print("You have ", calculate_pre_ovulation,"days left till the end of your safe period")
			elif safePeriod_input > 7 and safePeriod_input < post_ovulation_constant:
				calculate_post_ovulation = post_ovulation_constant -safePeriod_input
				print("You have ",calculate_post_ovulation, " days left to enter your safe period")
			elif safePeriod_input > 14 and safePeriod_input <= 28:
				calculate_post_ovulation2 = post_ovulation_constant2 - safePeriod_input
				print("You are currently on your safe period")
				print("You have ", calculate_post_ovulation2, " days left till the end of your period")
			else: 
				print("The day you entered is not within my expertise")
				print("I recommend you go see a doctor")
		case 3: 
			print(function_menstrual_app.in_app_display())
			ovulationTime_input = int(input('Enter the last day you saw your period: '))
			ovulation_check = calculate_ovulation_time(ovulationTime_input)
			print("Your next ovulation is in ", ovulation_check, " days time")
		case 0: print("Alright bye!")
		case _: print("Invalid input")
		
			
			
	

File: JAVA/MenstrualApp.java
import java.util.Scanner;
public class MenstrualApp{
static Scanner input = new Scanner(System.in);	
public static String displayWelcomePage(){
	String displayPage = """		
		Welcome to the Menstrual app!!!
		
Note: This app only works for regular cycle(28-day period)
		
		press 1 to enter if eligible """;
		
	return displayPage;}

public static String menuPage(){
	String displayMenu = """
				1. Flow date
				2. Safe period
				3. Ovulation time
				0. To exit the app
					""";
	return displayMenu;}
	
public static int calculateFlowDate(int dateInput){
	return 28 - dateInput;}

public static int calculateOvulationTime(int ovulationInput){
	int ovulationConstant = 14;
	int cycleConstant = 28;
	if(ovulationInput > 0 && ovulationInput <= ovulationConstant){
		return ovulationConstant - ovulationInput;}
	else if(ovulationInput > ovulationConstant && ovulationInput <=  cycleConstant){
		return cycleConstant - ovulationInput + ovulationConstant;
		}
	else
		return -1;
		}

public static String inAppDisplay(){
	String displayFlow = """
			Welcome!!!
			Enter the day you saw
			your period last
			 """;
	return displayFlow;
	}
				
public static void main(String[] args){	
	System.out.println(displayWelcomePage());
	System.out.println("Enter 1: ");
	 int user_input = input.nextInt();
	if(user_input == 1){
		System.out.println(menuPage());
		System.out.println("Enter an option: ");
		int menu_input = input.nextInt();
		switch(menu_input){
		case 1: System.out.println(inAppDisplay());
				System.out.println("Enter the last day you saw your period: ");
				int flowDateInput = input.nextInt();
				int flowCheck = calculateFlowDate(flowDateInput);
				System.out.println("Your next flow date is " + flowCheck + " days time");
				break;
		case 2: System.out.println(inAppDisplay());
				System.out.println("Enter the last day you saw your period: ");
				int safePeriodInput = input.nextInt();
				int preOvulationConstant = 7;
				int postOvulationConstant = 15;
				int postOvulationConstant2 = 28;
				if(safePeriodInput > 0 && safePeriodInput <= preOvulationConstant){
					int calculatePreOvulation = preOvulationConstant -  safePeriodInput;
					System.out.println("You are on your safe period");
					System.out.println("You have " + calculatePreOvulation + "days left till the end of your safe period");}
				else if(safePeriodInput > 7 && safePeriodInput < postOvulationConstant){
						int calculatePostOvulation = postOvulationConstant -safePeriodInput;
						System.out.println("You have " + calculatePostOvulation + " days left to enter your safe period");}
				else if(safePeriodInput > 14 && safePeriodInput <= 28){
					int calculatePostOvulation2 = postOvulationConstant2 - safePeriodInput;
						System.out.println("You are currently on your safe period");
						System.out.println("You have " + calculatePostOvulation2 + " days left till the end of your period");}
				else 
					System.out.printf("%s\n%s","The day you entered is not within my expertise", "I recommend you go see a doctor");
					break;
		case 3: System.out.println(inAppDisplay());
				System.out.println("Enter the last day you saw your period: ");
				int ovulationTimeInput =  input.nextInt();
				if(calculateOvulationTime(ovulationTimeInput) == -1){
					System.out.println("Invalid day");break;}
				int ovulationCheck = calculateOvulationTime(ovulationTimeInput);
					System.out.println("Your next ovulation is in " + ovulationCheck + " days time");
				break;	
		case 0: System.out.println("Alright bye!");
				break;
		default: System.out.println("Invalid input"); 
		}			
				}
			}
			}
			